src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.tsx

```diff
@@ -14,84 +14,212 @@
+  const [searchTrigger, result] = useLazyGetSpeciesSearchResultsQuery();
+  const { currentData, isLoading, isError } = result;
+
+  const {
+    genomes,
+    stagedGenomes,
+    isTableExpanded,
+    onTableExpandToggle,
+    onGenomePreselectToggle,
+    sortRule,
+    changeSortRule
+  } = useSelectableGenomesTable({
+    genomes: currentData?.matches ?? [],
+    selectedGenomes: selectedSpecies,
+    filterQuery
+  });
+
+  const deferredGenomes = useDeferredValue(genomes);
+
+  const maxSelectableGenomesCount =
+    'maxSelectableGenomesCount' in props
+      ? props.maxSelectableGenomesCount
+      : Infinity;
+  const selectedGenomesCount = selectedSpecies.length;
+
+  const onSearchInput = (event: FormEvent<HTMLInputElement>) => {
+    setSearchQuery(event.currentTarget.value);
+    if (!canSubmitSearch) {
+      setCanSubmitSearch(true);
     }
   };
 
-  const availableSpeciesList = popularSpeciesList.filter(
-    (species) => species.is_available
-  );
+  const onSpeciesAdd = () => {
+    props.onSpeciesAdd(stagedGenomes);
+  };
+
+  const onSearchSubmit = () => {
+    searchTrigger({ query: searchQuery });
+    setCanSubmitSearch(false);
+  };
 
   return (
-    <div className={styles.speciesSelector}>
-      <table className={styles.speciesSelectorTable}>
-        <thead>
-          <tr>
-            <th className={styles.commonNameHeader}>Common name</th>
-            <th className={styles.scientificNameHeader}>Scientific name</th>
-            <th className={styles.assemblyHeader}>Assembly</th>
-            <th className={styles.selectCol}>Select</th>
-          </tr>
-        </thead>
-        <tbody>
-          {availableSpeciesList.map((species, index) => {
-            return (
-              <tr key={index}>
-                <td>{species.common_name ?? '-'}</td>
-                <td className={styles.scientificNameCol}>
-                  {species.scientific_name}
-                </td>
-                <td className={styles.assemblyCol}>{species.assembly_name}</td>
-                <td className={styles.selectCol}>
-                  <Checkbox
-                    checked={selectedGenomeIds.includes(species.genome_id)}
-                    onChange={(isChecked) =>
-                      onSpeciesSelection(isChecked, species)
-                    }
-                  />
-                </td>
-              </tr>
-            );
-          })}
-        </tbody>
-      </table>
+    <div className={styles.main}>
+      <TopSection
+        query={searchQuery}
+        isLoading={isLoading}
+        isError={isError}
+        searchResults={currentData}
+        canAddGenomes={stagedGenomes.length > 0}
+        canSubmitSearch={canSubmitSearch}
+        onSearchSubmit={onSearchSubmit}
+        onSearchInput={onSearchInput}
+        onGenomesAdd={onSpeciesAdd}
+        onFilterChange={setFilterQuery}
+        onClose={onClose}
+      />
+
+      {currentData && currentData.matches.length > 0 && (
```

Comment:
Won't `currentData?.matches.length` make `currentData` check redundant?

Reply:
Typescript complains against `currentData?.matches.length > 0`; but since we are only checking for non-falsy length, `currentData?.matches.length` is indeed enough.

✅